auditbeat/tracing/int_aligned.go

```diff
@@ -20,14 +33,15 @@ func copyInt(dst unsafe.Pointer, src unsafe.Pointer, len uint8) error {
 	return nil
 }
 
-func readInt(ptr unsafe.Pointer, len uint8, signed bool) (value interface{}, err error) {
+func readInt(ptr unsafe.Pointer, len uint8, signed bool) (any, error) {
+	var value any
 	asSlice := (*(*[maxIntSizeBytes]byte)(ptr))[:]
```

Comment:
```suggestion
	asSlice := unsafe.Slice((*byte)(ptr), len)
```

Reply:
This could make use of parametric types so that `len` is not required and the type is known at compile time.

Reply:
hmmm trying to wrap my head around parametric types; All the fuzz in this function is because we try to determine the appropriate type based on the `len` and `signed` params. So even if we go with parametric types, aka generics, now the caller of readInt will have to determine the type and all the switch case fuzz will need to go there. Do I miss something?

Reply:
I was thinking of something like this https://go.dev/play/p/hcXbDct4Yj3

```
type integer interface {
	int8 | int16 | int32 | int64 | uint8 | uint16 | uint32 | uint64
}

func readInt[T integer](ptr unsafe.Pointer) T {
	var value T
	asSlice := unsafe.Slice((*byte)(ptr), unsafe.Sizeof(value))
	switch len(asSlice) {
	case 1:
		return T(asSlice[0])
	case 2:
		return T(binary.NativeEndian.Uint16(asSlice))
	case 4:
		return T(binary.NativeEndian.Uint32(asSlice))
	case 8:
		return T(binary.NativeEndian.Uint64(asSlice))
	default:
		panic("unreachable")
	}
}
```

Reply:
But yes, the switching would have to go elsewhere, so this is not useful.

Reply:
ok yes! this is similar to my thinking, the only minor diff I had in mind is the integer interface like this 
```
type integer interface {
	~int8 | ~int16 | ~int32 | ~int64 | ~uint8 | ~uint16 | ~uint32 | ~uint64
}
```

to allow also custom types that essentially are defined as integer primitives, but let's keep the switching here then. 👍